.travis.yml

```diff
@@ -0,0 +1,66 @@
+language: scala
+sudo: true
+services:
+  - docker
+
+before_install:
+  # make comparing to origin/master work and fetch full history for correct current and previous version detection
+  - git remote set-branches --add origin master && git fetch --unshallow
+  # using jabba for custom jdk management
+  - curl -sL https://raw.githubusercontent.com/shyiko/jabba/0.11.2/install.sh | bash && . ~/.jabba/jabba.sh
+
+# default script for jobs, that do not have any specified
+script:
+  - jabba install ${JDK:=adopt@~1.8-0}
+  - jabba use ${JDK:=adopt@~1.8-0}
+  - java -version
+  - sbt -jvm-opts .jvmopts-travis "$CMD"
+
+jobs:
+  include:
+    - stage: check
+      env: CMD="verifyCodeStyle"
+      name: "Code style check. Run locally with: sbt verifyCodeStyle"
+    - env: CMD="Test/compile"
+      name: "Compile all code with 2.13"
+
+    - stage: test
+      env:
+        - CMD="test"
+      name: "Run tests with Scala 2.13 and AdoptOpenJDK 8"
+    - env:
+        - JDK="adopt@~1.11-0"
+        - _JAVA_OPTIONS="-XX:+UnlockExperimentalVMOptions -XX:+UseJVMCICompiler"
+        - CMD="test"
+      name: "Run tests with Scala 2.13 and AdoptOpenJDK 11"
+
+    - stage: publish
+      env: CMD="publish"
+      name: "Publish artifacts to Bintray"
+
+
+stages:
+  # runs on master commits and PRs
+  - name: check
+    if: NOT tag =~ ^v
+
+  # runs on master commits and PRs
+  - name: test
+    if: NOT tag =~ ^v
+
+  # runs on main repo master commits and version-tagged commits
+  - name: publish
+    if: repo = akka/akka-persistence-jdbc AND ( ( branch = master AND type = push ) OR tag =~ ^v )
```

Comment:
Hmmm, publish on a push to `master`? (even if there's no tag)

Reply:
also the akka-persistence-jdbc leftover

Reply:
I'm comparing this to akka-grpc, it has `branch = master AND type != pull_request`. 

I guess what is happening here is that when you create a tag in GitHub, it's a push to master. So, either `type = push` or `type != pull_request`

Reply:
I'm comparing this to akka-grpc, it has `branch = master AND type != pull_request`.

I guess what is happening here is that when you create a tag in GitHub, it's a push to master. So, either `type = push` or `type != pull_request`

Reply:
Hmm, pushing a tag `upstream` is a push to `master`? What if the tag is on  stable branch `1.5.x`?

Reply:
Ah, you mean `when you create a tag in GitHub` in GitHub in particular... I see. 🤔 I have no idea.

Reply:
I still find somethings weird on the conditon `( ( branch = master AND type = push ) OR tag =~ ^v )` but if you copied it from elsewhere and it works there we can go ahead and fix it later.

Reply:
Yes, that's what I mean. But I think you spotted something that we were not aware of. 

This is an OR condition. So, either we are publishing because adding a tag triggers a push on master (which is what I think it's happening) or we publish because there is a tag (which explain how we can publish for other branches).

I took it from akka-persistence-jdbc. It was added by Ennu. Alpakka projects are doing the same and akka-grpc as well. 

@ennru / @raboof, can you shed a light on it? 

I think the right condition should be: 
`if: repo = akka/akka-projection AND tag =~ ^v`

Reply:
`master` is there to do the snapshot releases on every merge to master.

Reply:
@ennru that needs a bit more setting up, right? I mean, we'll probably need `sbt-dynver`, maybe some extra setting in the publishing logic?

BTW, I'm 100% in favor of `sbt-dynver` and publishing _all_ commits in `master` to a snapshots repo.